src/software/ai/navigator/obstacle/robot_navigation_obstacle_factory.cpp

```diff
@@ -199,3 +200,46 @@ ObstaclePtr RobotNavigationObstacleFactory::createFromFieldRectangle(
     return std::make_shared<GeomObstacle<Polygon>>(
         Rectangle(Point(xMin, yMin), Point(xMax, yMax)));
 }
+
+ObstaclePtr RobotNavigationObstacleFactory::createFromBallPlacement(
+    const Point &placement_point, const Point &ball_point) const
+{
+    /*   The Polygon is constructed as follows:
+     *
+     *         ball_l           ball_r
+     *           +-------+-------+
+     *           |               |
+     *           |    ball_c     |
+     *           +-------+-------+
+     *           |               |
+     *           |               |
+     *           |               |
+     *           |     place_c   |
+     *           +-------+-------+
+     *           |               |
+     *           |               |
+     *           +-------+-------+
+     *        place_l          place_r
+     */
+    const double RADIUS = 0.5;
+
+    Vector place_to_ball = placement_point.toVector() - ball_point.toVector();
+    Vector ball_to_place = ball_point.toVector() - placement_point.toVector();
```

Comment:
these are opposite vectors, so we can simplify the code by just calculating one of them and flipping +/-